packages/rln/src/contract/price_calculator.test.ts

```diff
@@ -0,0 +1,36 @@
+import { expect } from "chai";
+import { ethers } from "ethers";
+
+import { RLNBaseContract } from "./rln_base_contract.js";
+
+function createRealRLNBaseContract(
+  provider: ethers.providers.Provider
+): RLNBaseContract {
+  const dummy = Object.create(RLNBaseContract.prototype);
+  dummy.contract = { provider };
+  return dummy as RLNBaseContract;
+}
+
+describe("RLNBaseContract.getPriceForRateLimit (integration)", function () {
+  this.timeout(10000);
+
+  const rpcUrl = "https://rpc.sepolia.linea.build";
+  let provider: ethers.providers.JsonRpcProvider;
+
+  before(() => {
+    provider = new ethers.providers.JsonRpcProvider(rpcUrl);
```

Comment:
this seems to be integrational test: the one that checks that API is getting called
if it is then there are some things that must be done differently

- overall UTs must not send any HTTP requests to outside API, this one, as it seems to me, will use RPC you provided, this is wrong for UT in general

what should be done is:
- mock for contract need to be created
- mock for `RLNBaseContract`
- test should cover: if `.calculate` is called and what `getPriceForRateLimit` returns based on different behavior of `calculate`

let me know if something is not clear

Reply:
Resolved by 8be09783c423ead85f9531c346cc3339fbcf4c7e